sources/github.go

```diff
@@ -652,6 +653,30 @@ func (g *githubSource) GetDefaultBranch(ctx context.Context, accessToken *Access
 	return *gitRepo.DefaultBranch, nil
 }
 
+func (g *githubSource) WaitForCommit(ctx context.Context, accessToken *AccessToken, owner, repo, message string) error {
+	githubClient := g.interactionsFunc(ctx, accessToken.Token, accessToken.Type, g.cfg.RateLimitTimeoutSeconds, g.cfg.RateLimitRetryCount)
+
+	err := retry.Retry(time.Duration(g.cfg.WaitTagTimeoutSeconds)*time.Second, func(i int) error {
+		ref, _, err := githubClient.GetRepoRef(ctx, owner, repo, "heads/main")
+		if err != nil {
+			return err
+		}
+
+		commit, err := githubClient.GetCommit(ctx, owner, repo, *ref.Object.SHA)
+		if err != nil {
+			return err
+		}
+
+		if *commit.Message != message {
+			return errors.Wrapf(ErrCommitNotFound, "last commit is not %s", message)
+		}
+
+		return nil
+	})
+
+	return err
+}
```

Comment:
How do we know that this will actually get us the latest commit? what happens if I do multiple commits to the main branch, but not all of them have been updated remotely yet? Could I get a ref that is not the latest commit?

Reply:
It is the latest commit on remote. We don't care if the latest is the latest we pushed, that's why we retry (we wait for it to show up) and search for the message we think is the latest one.

Reply:
Makes sense, but matching by commit message is not ideal - you could have a quick series of commits with the same message and different contents or you can have someone outside your scope committing to the repo with the same message.